test/e2e/case15_event_format_test.go

```diff
@@ -0,0 +1,170 @@
+		Expect(parent).NotTo(BeNil())
+
+		By("Creating compliant policy " + case15AlwaysCompliantName + " on " + testNamespace + " with parent " + case15AlwaysCompliantParentName)
+		plcDef := utils.ParseYaml(case15AlwaysCompliantYaml)
+		ownerRefs := plcDef.GetOwnerReferences()
+		ownerRefs[0].UID = parent.GetUID()
+		plcDef.SetOwnerReferences(ownerRefs)
+		_, err := clientManagedDynamic.Resource(gvrConfigPolicy).Namespace(testNamespace).Create(context.TODO(), plcDef, v1.CreateOptions{})
+		Expect(err).To(BeNil())
+
+		plc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy, case15AlwaysCompliantName, testNamespace, true, defaultTimeoutSeconds)
+		Expect(plc).NotTo(BeNil())
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy, case15AlwaysCompliantName, testNamespace, true, defaultTimeoutSeconds)
+			return utils.GetComplianceState(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+
+		By("Checking events on the configurationpolicy")
+		compPlcEvents := utils.GetMatchingEvents(clientManaged, testNamespace, case15AlwaysCompliantName, "", "Policy status is: Compliant", defaultTimeoutSeconds)
+		Expect(compPlcEvents).NotTo(BeEmpty())
+		nonCompPlcEvents := utils.GetMatchingEvents(clientManaged, testNamespace, case15AlwaysCompliantName, "", "Policy status is: NonCompliant", defaultTimeoutSeconds)
+		Expect(nonCompPlcEvents).To(BeEmpty())
+
+		By("Checking events on the parent policy")
+		compParentEvents := utils.GetMatchingEvents(clientManaged, testNamespace, case15AlwaysCompliantParentName, "policy: "+testNamespace+"/"+case15AlwaysCompliantName, "^Compliant;", defaultTimeoutSeconds)
+		Expect(compParentEvents).NotTo(BeEmpty())
+		nonCompParentEvents := utils.GetMatchingEvents(clientManaged, testNamespace, case15AlwaysCompliantParentName, "policy: "+testNamespace+"/"+case15AlwaysCompliantName, "^NonCompliant;", defaultTimeoutSeconds)
+		Expect(nonCompParentEvents).To(BeEmpty())
+	})
+	It("Records the right events for a policy that is never compliant", func() {
+		By("Creating parent policy " + case15NeverCompliantParentName + " on " + testNamespace)
+		utils.Kubectl("apply", "-f", case15NeverCompliantParentYaml, "-n", testNamespace)
+		parent := utils.GetWithTimeout(clientManagedDynamic, gvrPolicy, case15NeverCompliantParentName, testNamespace, true, defaultTimeoutSeconds)
+		Expect(parent).NotTo(BeNil())
+
+		By("Creating noncompliant policy " + case15NeverCompliantName + " on " + testNamespace + " with parent " + case15NeverCompliantParentName)
+		plcDef := utils.ParseYaml(case15NeverCompliantYaml)
+		ownerRefs := plcDef.GetOwnerReferences()
+		ownerRefs[0].UID = parent.GetUID()
+		plcDef.SetOwnerReferences(ownerRefs)
+		_, err := clientManagedDynamic.Resource(gvrConfigPolicy).Namespace(testNamespace).Create(context.TODO(), plcDef, v1.CreateOptions{})
+		Expect(err).To(BeNil())
```

Comment:
Optional: These two `By` sections seem pretty similar to the other tests. You could create a helper function to perform this by accepting the names and YAML paths of the parent and config policy.

Reply:
In my opinion it doesn't matter if it's all localized to one file or not. It's still duplicated several times.  :)

Though, I won't block on it since it's in the tests.